components/server/src/workspace/workspace-starter.ts

```diff
@@ -383,6 +394,28 @@ export class WorkspaceStarter {
         }
     }
 
+    private ensureShutdown(
+        ctx: TraceContext,
+        shutdownToken: CancellationToken,
+        workspace: Workspace,
+        instance?: WorkspaceInstance,
+    ) {
+        if (!shutdownToken.isCancellationRequested) {
+            return;
+        }
+        const shutdownError = this.failOnShutdown(ctx, workspace, instance);
+        throw new StartInstanceError("serverShuttingDown", shutdownError);
+    }
+    private failOnShutdown(ctx: TraceContext, workspace: Workspace, instance?: WorkspaceInstance) {
+        const shutdownError = new Error("Service temporarily unavailable. Please retry in a few seconds.");
```

Comment:
`Service temporarily unavailable. Please retry **to start** in a few seconds again.`